internal/translayer/dbuser/dbuser.go

```diff
@@ -0,0 +1,74 @@
+package dbuser
+
+import (
+	"context"
+	"errors"
+
+	"go.mongodb.org/atlas-sdk/v20231115008/admin"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/internal/translayer"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/atlas"
+)
+
+var (
+	// ErrorNotFound is returned when an database user was not found
+	ErrorNotFound = errors.New("database user not found")
+)
+
+type Service struct {
+	admin.DatabaseUsersApi
```

Comment:
I am sceptical in embedding a foreign API `admin.DatabaseUsersApi` and thus exposing all of its methods here as it easily allows bypassing the whole service layer. Is it intentional?

Reply:
coming in next update